test/services/AppointRevokeAgentServiceSpec.scala

```diff
@@ -0,0 +1,129 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import connectors.PropertyRepresentationConnector
+import connectors.propertyLinking.PropertyLinkConnector
+import models._
+import models.searchApi.OwnerAuthAgent
+import repositories.SessionRepo
+import uk.gov.hmrc.http.HeaderCarrier
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import org.scalatest.concurrent.ScalaFutures
+import uk.gov.hmrc.http.logging.SessionId
+
+import scala.concurrent.Future
+
+class AppointRevokeAgentServiceSpec extends ServiceSpec {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  private lazy val testService = new AppointRevokeAgentService(mockRepresentationConnector, mockPropertyLinkConnector, mockSessionRepo)
+
+  private lazy val mockPropertyLinkConnector = mock[PropertyLinkConnector]
+
+  private lazy val mockRepresentationConnector = mock[PropertyRepresentationConnector]
+
+  private lazy val mockSessionRepo = mock[SessionRepo]
+
+  implicit val hc = HeaderCarrier(sessionId = Some(SessionId("1111")))
+
+
+
+  "createAndSubmitAgentRepRequest" should "return option unit when succesful" in {
+
+    val links = SessionPropertyLinks(Seq(SessionPropertyLink(1L, "1", Seq(OwnerAuthAgent(1l, 1l, "", None,
+      AgentPermission.fromName("START_AND_CONTINUE").get, AgentPermission.fromName("START_AND_CONTINUE").get)))))
+
+    when(mockSessionRepo.get[SessionPropertyLinks](any(), any())).thenReturn(Future.successful(Some(links)))
+    when(mockRepresentationConnector.revoke(any())(any())).thenReturn(Future.successful())
+    when(mockRepresentationConnector.create(any())(any())).thenReturn(Future.successful())
+
+
+    val res = testService.createAndSubmitAgentRepRequest(
+      List("1"),
+      1L,
+      1L,
+      1L,
+      AgentPermission.fromName("START_AND_CONTINUE").get,
+      AgentPermission.fromName("START_AND_CONTINUE").get,
+      true)
+
+    res.futureValue must be(())
+
+  }
+
+
+  "createAndSubmitAgentRepRequest" should "throw exception when link doesn't exist in cache" in {
+
+    val links = SessionPropertyLinks(Seq(SessionPropertyLink(2L, "11111", Seq(OwnerAuthAgent(3L, 4L, "", None,
+      AgentPermission.fromName("START_AND_CONTINUE").get, AgentPermission.fromName("START_AND_CONTINUE").get)))))
+
+    when(mockSessionRepo.get[SessionPropertyLinks](any(), any())).thenReturn(Future.successful(Some(links)))
+
+    val res = testService.createAndSubmitAgentRepRequest(
+      List("999", "8888"),
+      5L,
+      6L,
+      7L,
+      AgentPermission.fromName("START_AND_CONTINUE").get,
```

Comment:
you should just use AgentPermission.START_AND_CONTINUE. this operation would break if the permissions values changed.

Reply:
done